srcmedia/ts/components/PageSearchForm.ts

```diff
@@ -0,0 +1,42 @@
+import { Subject } from 'rxjs'
+
+import { Reactive } from '../lib/common'
+import { RxForm, RxFormState } from '../lib/form'
+import { ajax } from '../../js/modules/Utilities'
+
+interface PageSearchFormState extends RxFormState {
+    results: string
+}
+
+class PageSearchForm extends RxForm implements Reactive<PageSearchFormState>{
+    state: Subject<PageSearchFormState>
+
+    constructor(element: HTMLFormElement) {
+        super(element)
+        this.state = new Subject()
+        this.submit = this.submit.bind(this) // so it can be called externally
+    }
+
+    /**
+     * Serialize the form and submit it as a GET request to the form's endpoint,
+     * passing the response to update().
+     * 
+     * Also updates the browser history, saving the search.
+     *
+     * @returns {Promise<void>}
+     * @memberof PageSearchForm
+     */
+    async submit(): Promise<void> {
+        return fetch(`${this.target}?${this.serialize()}`, ajax)
+            .then(res => res.text())
+            .then(html => this.update({ results: html }))
+            .then(() => window.history.pushState(null, 'PPA Archive Search', this.serialize()))
```

Comment:
I think the pushed state is missing the `?` in the url - when I tested running it, the ajax form works but when I press enter I get a bad url: `archive/mdp.39015009444814/query%3Dinsect/?query=insect`

Related: should we disable submitting the form on enter?

Additionally: we shouldn't hardcode `PPA Archive Search` in the history; can you grab the html title from the page?

Reply:
I was running into this too! I thought it was some kind of encoding thing but I think you're right, just needs the `?`.

Makes sense to disable enter to submit & agree about grabbing the title, seems like a good idea.